examples/rust/custom_space_view/src/color_coordinates_space_view.rs

```diff
@@ -0,0 +1,248 @@
+    // State type as described above.
+    type State = ColorCoordinatesSpaceViewState;
+
+    // Scene context is shared between all scene parts.
+    // For this Space View we don't need any scene context.
+    type Context = ();
+
+    // Collection of scene parts that are needed to display a frame.
+    // We only have a single scene part, but it could be a collection of various.
+    type SceneParts = SceneColorCoordinates;
+
+    // Scene parts can have a common data object that they expose.
+    // For this Space View this is not needed.
+    type ScenePartData = ();
+
+    fn name(&self) -> SpaceViewClassName {
+        // Name and identifier of this Space View.
+        "Color Coordinates".into()
+    }
+
+    fn icon(&self) -> &'static re_ui::Icon {
+        &re_ui::icons::SPACE_VIEW_SCATTERPLOT
+    }
+
+    fn help_text(&self, _re_ui: &re_ui::ReUi, _state: &Self::State) -> egui::WidgetText {
+        "A demo space view that shows colors as coordinates on a 2D plane.".into()
+    }
+
+    fn preferred_tile_aspect_ratio(&self, _state: &Self::State) -> Option<f32> {
+        // Prefer a square tile if possible.
+        Some(1.0)
+    }
+
+    /// Additional UI displayed when the space view is selected.
+    ///
+    /// In this sample we show a combo box to select the color coordinates mode.
+    fn selection_ui(
+        &self,
+        _ctx: &mut ViewerContext<'_>,
+        ui: &mut egui::Ui,
+        state: &mut Self::State,
+        _space_origin: &EntityPath,
+        _space_view_id: SpaceViewId,
+    ) {
+        ui.horizontal(|ui| {
+            ui.label("Coordinates mode");
+            egui::ComboBox::from_id_source("color_coordinates_mode")
+                .selected_text(state.mode.to_string())
+                .show_ui(ui, |ui| {
+                    ui.style_mut().wrap = Some(false);
+                    ui.set_min_width(64.0);
+
+                    let mut selectable_value = |ui: &mut egui::Ui, e| {
+                        ui.selectable_value(&mut state.mode, e, e.to_string())
+                    };
+                    selectable_value(ui, ColorCoordinatesMode::Hs);
+                    selectable_value(ui, ColorCoordinatesMode::Hv);
+                    selectable_value(ui, ColorCoordinatesMode::Rg);
+                });
+        });
+    }
+
+    /// The contents of the Space View window and all interaction within it.
+    ///
+    /// This is called with a fully populated scene which is a combination of all
+    /// scene contexts, parts and selection highlights.
+    fn ui(
+        &self,
+        ctx: &mut ViewerContext<'_>,
+        ui: &mut egui::Ui,
+        state: &mut Self::State,
+        scene: &mut TypedScene<Self>,
+        _space_origin: &EntityPath,
+        space_view_id: SpaceViewId,
+    ) {
+        egui::Frame {
+            inner_margin: re_ui::ReUi::view_padding().into(),
+            ..egui::Frame::default()
+        }
+        .show(ui, |ui| {
```

Comment:
We usually only add padding around spaceviews that contain text, buttons etc. Our spatial space views go all the way to the edge